File: src/main/java/com/hackanton/event/Event.java
package com.hackanton.event;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;

/**
 * @author ddorochov
 */
public interface Event {

	@NotNull
	String getTitle();

	@NotNull
	EventType getType();

	@NotNull
	List<String> getTopics();

	@NotNull
	PriceType getPriceType();

	/**
	 * @return date and time of this event
	 */
	@NotNull
	LocalDateTime getDateAndTime();

	/**
	 * @return page from which this event was parsed
	 */
	@NotNull
	String getPageUrl();

	@NotNull
	EventAddress getAddress();

	@Nullable
	EventAgenda getAgenda();

	@NotNull
	String getDescription();

	@NotNull
	Language getLanguage();
}

File: app/src/main/java/com/hamboomger/model/event/Event.kt
package com.hamboomger.model.event

import com.hamboomger.model.common.Language
import java.time.LocalDateTime
import java.util.*
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.MappedSuperclass

/**
 * @author ddorochov
 */
@MappedSuperclass
open class Event(
    val type: EventType,
    val topics : List<String>,
    val priceType: PriceType,
    val dateAndTime: LocalDateTime,
    val pageUrl: String,
    val address: EventAddress,
    val agenda: EventAgenda,
    val description: String,
    val language: Language
) {
    @Id
    @GeneratedValue
    private val id = 0L
}